pipeline/post-results-new.groovy

```diff
@@ -0,0 +1,278 @@
+                println("Stage : ${stageLevel}")
+                def tier_index = tags_list.findIndexOf { it ==~ /tier-\w+/ }
+                tierLevel = tags_list.get(tier_index)
+                println("Tier : ${tierLevel}")
+                run_type = msgMap["pipeline"]["run_type"]
+                println("Type : ${run_type}")
+                build_url = msgMap["run"]["url"]
+                println("Build : ${build_url}")
+            }
+        }
+
+        stage('configureReportPortalWorkDir') {
+            println("Stage configureReportPortalWorkDir")
+            (rpPreprocDir, credsRpProc) = reportLib.configureRpPreProc(sharedLib)
+        }
+
+        stage('uploadTestResultToReportPortal'){
+            println("Stage uploadTestResultToReportPortal")
+            composeInfo = msgMap["recipe"]
+
+            def resultDir = msgMap["test"]["object-prefix"]
+            println("Test results are available at ${resultDir}")
+
+            def tmpDir = sh(returnStdout: true, script: "mktemp -d").trim()
+            tags_list = msgMap["pipeline"]["tags"].split(',') as List
+            if ('ibmc' in tags_list) {
+                sh script: "rclone sync ${remoteName}://${reportBucket} ${tmpDir} --progress --create-empty-src-dirs"
+            } else {
+                sh "sudo cp -r /ceph/cephci-jenkins/${resultDir} ${tmpDir}"
+            }
+
+            metaData = readYaml file: "${tmpDir}/${resultDir}/metadata.yaml"
+            println("metadata: ${metaData}")
+            def copyFiles = "cp -a ${tmpDir}/${resultDir}/results ${rpPreprocDir}/payload/"
+            def copyAttachments = "cp -a ${tmpDir}/${resultDir}/attachments ${rpPreprocDir}/payload/"
+            def rmTmpDir = "rm -rf ${tmpDir}"
+
+            // Modifications to reuse methods
+            println("modifying metaData")
+            metaData["ceph_version"] = metaData["ceph-version"]
+            if ( metaData["stage"] == "latest" ) { metaData["stage"] = "Tier-0" }
+
+            println("copyFiles")
+            sh script: "${copyFiles} && ${copyAttachments} && ${rmTmpDir}"
+
+            println("buildArtifacts")
+            if ( composeInfo ){
+                metaData["buildArtifacts"] = composeInfo
+            }
+
+            println("uploading results")
+            if (metaData["results"]) {
+                def launch_id = reportLib.uploadTestResults(rpPreprocDir, credsRpProc, metaData, stageLevel, run_type)
+                if (launch_id) {
+                    metaData["rp_link"] = "${rp_base_link}/ui/#cephci/launches/all/${launch_id}"
+                }
+            }
+
+            // Remove the sync results folder
+            if ('ibmc' in tags_list) {
+                sh script: "rclone purge ${remoteName}:${reportBucket}/${resultDir}"
+            } else {
+                sh "sudo rm -r /ceph/cephci-jenkins/${resultDir}"
+            }
+        }
+
+        stage('notifyTier-0Failure'){
```

Comment:
`notifyTier-0Failure` ???

Reply:
Stage is to notify through when a tier-0 execution fails.
Unable to add skip stage logic,
-  tried using the option when, 
- Also tried manually setting stage status
none of the above worked. 
Will work on this as part of a different PR